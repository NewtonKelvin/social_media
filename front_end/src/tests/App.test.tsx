import {
  fireEvent,
  getByLabelText,
  render,
  waitFor,
} from "@testing-library/react";
import Login from "../pages/login";
import { screen } from "@testing-library/dom";

test("Test", () => {
  expect(1 + 1).toBe(2);
});

test("Login page", async () => {
  render(<Login />);

  const username = screen.getByLabelText("Username:");
  const password = screen.getByLabelText("Password:");
  expect(username).toBeInTheDocument();
  expect(password).toBeInTheDocument();
});

test("Login action", async () => {
  const container = render(<Login />);

  const username = screen.getByLabelText("Username:");
  const password = screen.getByLabelText("Password:");
  const submit = screen.getByRole("button", { name: "Login" });

  fireEvent.change(username, { target: { value: "kelvin.newton" } });
  fireEvent.change(password, { target: { value: "123123" } });
  //fireEvent.click(submit);
});
