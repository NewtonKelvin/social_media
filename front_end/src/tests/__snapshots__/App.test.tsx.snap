// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login action 2`] = `[Function]`;

exports[`Login action 3`] = `[Function]`;

exports[`Login action 4`] = `[Function]`;

exports[`Login action 5`] = `[Function]`;

exports[`Login action 6`] = `[Function]`;

exports[`Login action 7`] = `[Function]`;

exports[`Login action 8`] = `[Function]`;

exports[`Login action 9`] = `[Function]`;

exports[`Login action 10`] = `[Function]`;

exports[`Login action 11`] = `[Function]`;

exports[`Login action 12`] = `[Function]`;

exports[`Login action 13`] = `[Function]`;

exports[`Login action 14`] = `[Function]`;

exports[`Login action 15`] = `[Function]`;

exports[`Login action 16`] = `[Function]`;

exports[`Login action 17`] = `[Function]`;

exports[`Login action 18`] = `[Function]`;
